src/core/io/src/4C_io_meshreader.cpp

```diff
@@ -138,9 +138,22 @@ void Core::IO::MeshReader::rebalance()
     Teuchos::ParameterList rebalanceParams;
     rebalanceParams.set<std::string>("imbalance tol", std::to_string(imbalance_tol));
 
+    const int minele_per_proc = parameters_.mesh_paritioning_parameters.get<int>("MIN_ELEPROC");
+    const int max_global_procs = Core::Communication::num_mpi_ranks(comm_);
+    int min_global_procs = max_global_procs;
+
+    if (minele_per_proc > 0)
+      min_global_procs =
+          element_readers_[i].get_row_elements()->NumGlobalElements() / minele_per_proc;
+    const int procs = min_global_procs > max_global_procs ? max_global_procs : min_global_procs;
+    rebalanceParams.set<std::string>("num parts", std::to_string(procs));
+
     const auto rebalanceMethod = Teuchos::getIntegralValue<Core::Rebalance::RebalanceType>(
         parameters_.mesh_paritioning_parameters, "METHOD");
 
+    if (!Core::Communication::my_mpi_rank(comm_))
+      std::cout << "\nProcs used for redistribution: " << procs << "\n";
```

Comment:
The screen output sounds a bit like it will be listing the IDs of the process. Rephrase?
```suggestion
    if (!Core::Communication::my_mpi_rank(comm_))
      std::cout << "\nNumber of procs used for redistribution: " << procs << "\n";
```
or
```suggestion
    if (!Core::Communication::my_mpi_rank(comm_))
      std::cout << "\nDistributing discretization to " << num_procs << " MPI ranks...\n";
```